text/0000-multi-api-lifecycle-descriptor.md

```diff
@@ -0,0 +1,172 @@
+# Meta
+[meta]: #meta
+- Name: Lifecycle Descriptor with Multiple APIs
+- Start Date: 2020-07-02
+- Author(s): ekcasey
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [RFC-0011](https://github.com/buildpacks/rfcs/blob/main/text/0011-lifecycle-descriptor.md)
+
+# Summary
+[summary]: #summary
+
+This RFC proposed three levels of API support: `deprecated`, `supported`, and `experimental`. Supported APIs will be expressed in ranges.
+
+This RFC introduces a new lifecycle descriptor file format, and an analogous label that should be applied to builder and lifecycle images.
+
+# Motivation
+[motivation]: #motivation
+
+Now that the lifecycle may simultaneously support multiple APIs, the current lifecycle descriptor format, which lists exactly one platform and one buildpack API, is no longer sufficient.
+
+In addition, we should have a mechanism for deprecating an entire API, so that buildpack and platform authors are not caught off guard when their platform or buildpack no longer works with a newer lifecycle. We likely will not choose to deprecate any APIs in the near future but building API deprecation into any new design prepares us well for the future.
+
+Experimental APIs will give platform and buildpack authors an opportunity to try out unreleased APIs and give us feedback. This will be valuable as we prepare for the `1.0` versions of the APIs. We want to get the design of the `1.0` APIs as close to "right" as possible because `1.n` APIs will not permit breaking changes. Feedback will help us improve the design before we finalize it. The quality of feedback we get from user about their real life usage of experimental APIs will help us avoid problems we may miss when approving RFCs.
+
+# What it is
+[what-it-is]: #what-it-is
+
+## Lifecycle Descriptor
+The following is an example `lifecycle.toml` following the proposed schema:
```

Comment:
I'm confused about how/where this file is used.

Reply:
`pack create-builder` uses it to set metadata on the builder and to validate that a builder has compatible lifecycle/buildpacks.

Reply:
does that mean `pack create-builder` would have two files as input (with `builder.toml` being the second)? Could those two files be merged?

Reply:
`pack` when creating a builder expects to find the `lifecycle.toml` inside the lifecycle blob. The user doesn't need to specify it.